### PR TITLE
Reuse CancellationForm in OfferPresenter + Widget

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- See title.

- Remove existing implementation of cancellation form and related functions

## Justify why they are needed

- Collect as much logic about the cancellation form as possible in one place

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3796?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #3787** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3787?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3795** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3795?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3796** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3796?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3796?utm_source=stack-comment).

**Review by gustaveen — APPROVED** [review]
Great work! 🙌 We could add some stories for some of the components later as well

File: apps/store/src/components/Cancellation/BankSigneringCancellation.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import { useFormatter } from '@/utils/useFormatter'
+import { AutoSwitchToggleCard } from './AutoSwitchToggleCard'
+
+type Props = {
+  startDate?: Date
+  onStartDateChange: (date: Date) => void
+  requested: boolean
+  onAutoSwitchChange?: (checked: boolean) => void
+  companyName: string
+  loading?: boolean
```
Comment (guilhermespopolin) [inline apps/store/src/components/Cancellation/BankSigneringCancellation.tsx:13]: nit: This seems to be not used

File: apps/store/src/components/Cancellation/BankSigneringInvalidRenewalDateCancellation.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { useTranslation } from 'next-i18next'
+import { Space } from 'ui'
+import { InputStartDay } from '@/components/InputDay/InputStartDay'
+import { useFormatter } from '@/utils/useFormatter'
+import { InfoCard } from '../InfoCard/InfoCard'
+
+type Props = {
+  startDate: Date
+  onStartDateChange: (date: Date) => void
+  loading?: boolean
```
Comment (guilhermespopolin) [inline apps/store/src/components/Cancellation/BankSigneringInvalidRenewalDateCancellation.tsx:10]: nit: same here; seems not to be used

**Review by guilhermespopolin — APPROVED** [review]
Well done 🚀 
Just left a couple of _nit_ comments.

**Comment by robinandeer** [comment]
### Merge activity

* **Jan 3, 2:28 PM**: @robinandeer started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3796).

**Pull request merged**